nlp_primitives/total_word_length.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+
+from featuretools.primitives.base import TransformPrimitive
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Double, NaturalLanguage
+
+
+class TotalWordLength(TransformPrimitive):
+    """Determines the total word length.
+
+    Description:
+        Given list of strings, determine the total
+        word length in each string. A word is defined as
+        a series of any characters not separated by a delimiter.
+        If a string is empty or `NaN`, return `NaN`.
+
+    Args:
+        delimiters_regex (str): Delimiters as a regex strign to use for splitting strings into words.
+            The default delimiters include "- [].,!?;\\n]".
+
+    Examples:
+        >>> x = ['This is a test file', 'This is second line', 'third line $1,000', None]
+        >>> total_word_length = TotalWordLength()
+        >>> total_word_length(x).tolist()
+        [15.0, 16.0, 14.0, nan]
+    """
+    name = "total_word_length"
+    input_types = [ColumnSchema(logical_type=NaturalLanguage)]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={'numeric'})
```

Comment:
In this implementation, if the series does not contain null values, it'll return an integer dtype otherwise float. If we want to return floats, then we'll need to convert the series dtype in those cases.

Reply:
I don't think this should matter much. By specifying the return type as `Double` here, things will get converted to a float when Woodwork is initialized on the feature matrix.

Reply:
If type conversions happen only after applying all the primitives, aggregation primitives can get unexpected inputs types. For example, an aggregation primitive after this transform primitive can get an integer type in some cases even when the input type is double.

Reply:
Yeah, technically I think that could be an issue, but as of now I don't think we have any primitives that specify a specific numeric logical type (Double, Integer, IntegerNullable, etc) as input, do we? I think everything that takes in a numeric value specifies the input like this: 
```
input_types = [ColumnSchema(semantic_tags={'numeric'})]
```

If we don't know the logical type exactly, you can specify the return type without the logical type, which is probably more consistent with the actual behavior in this case:
```
return_type = ColumnSchema(semantic_tags="numeric")
```

Reply:
Yeah, I'm mainly just highlighting the unexpected input types. It doesn't have to be numeric. If a transform primitive returns an unknown type when the return type is double, then that could raise issues.